Add code documentation

JakubBilski opened the pull request:
I followed this www.datacamp.com/community/tutorials/docstrings-python tutorial.
What I wrote can be used to automatically create a documentation in html format.
Example (our main.py):
![docsSample](https://user-images.githubusercontent.com/47048420/101081645-9f676d00-35aa-11eb-9f96-391877c2ec73.jpg)
I updated readme with instructions on how to use this generator

JakubBrojacz commented on src/music/tonation.py:
```diff
@@ -3,6 +3,18 @@
+
+    Attributes:
+        kind (str) : "minor" or "major"
+        tonic (Chord) : Set of sounds composing the
```
I feel like we can delete toniv, subdominant and dominant params. It is only used in bar_algorithm and we can remove it (or even leave just daria)

JakubBilski replied:
Removed

JakubBilski commented on src/sounds_manipulation.py:
```diff
@@ -2,6 +2,17 @@
+    Parameters:
+    sounds (list[Sound]) : Sounds to transpose
+    diff (int) : Chosen number of halftones
+    duration_ms_of_32 (float) : Duration of 32th note in miliseconds
```
Fixed

JakubBilski commented on src/main.py:
```diff
@@ -48,6 +48,7 @@ def parse_args():
 
 @app.route('/recorded', methods=['POST'])
 def frontend_communication_upload_recorded():
+    """Used to upload a previously recorded audio file"""
```
Added, I didn't go too much into details, hope it's enough

JakubBilski commented on src/chords_simplification.py:
```diff
@@ -38,6 +38,22 @@
 
 def simplify(notes, chords, tonation, preview_file):
+    """Transpose all piece information to a key
+    in which it should be easier to play
```
the piece
I rewrote it, I think it's better now

JakubBilski commented on src/chords_simplification.py:
```diff
@@ -38,6 +38,22 @@
+    (list[Sound]) : Notes after transposition
+    (list[Chord]) : Chords after transposition
+    (Tonation) : Chosen easier key
+    (os.path-like) : Same as input preview_file
```
Fixed

JakubBilski commented on src/chords_generation/daria_params.py:
```diff
@@ -1,3 +1,4 @@
+"""Global arameters used by chords generation functions"""
```
Fixed

JakubBilski commented:
Ok, so it turns out, that if something is a part of a package, you should never run it as a script.
https://stackoverflow.com/questions/16981921/relative-imports-in-python-3
That's the whole reason it was so hard to get tests to work, and thet's the reason we needed path_magic. I moved main and main_tests outside the src folder, and changed a lot of imports (if not all of them) to relative. Now the whole app still works, we don't need path_magic, and authomatic documentation also works (it's in the html folder).
I would love to change music_server.py name to sth else, but I didn't have a good idea.
